projects/media-viewer/src/lib/media-viewer/media-viewer.model.ts

```diff
@@ -1,8 +1,10 @@
 import { Subject } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
 
 export class ActionEvents {
   public readonly rotate = new Subject<RotateOperation>();
-  public readonly search = new Subject<SearchOperation>();
+  public readonly search = new Subject<SearchOperation>().pipe(debounceTime(250));
```

Comment:
why do we need the debounceTime?

Reply:
the event is triggered on every keypress and the PDF lib struggles to keep up when you type quickly